src/test/java/mate/academy/security/AuthenticationServiceImplTest.java

```diff
@@ -0,0 +1,125 @@
+        authenticationService = new AuthenticationServiceImpl(userService, roleService, passwordEncoder);
+    }
+
+    @Test
+    void register_OK() {
+        User vitalii = new User();
+        String email = "vitalii@gmail.com";
+        String password = "1234";
+        Role vitalisRole = new Role(Role.RoleName.USER);
+        vitalii.setEmail(email);
+        vitalii.setPassword(password);
+        Mockito.when(roleService.getRoleByName(any())).thenReturn(vitalisRole);
+        Mockito.when(userService.save(any())).thenReturn(vitalii);
+        User actualUser = authenticationService.register(vitalii.getEmail(), vitalii.getPassword());
+        Assertions.assertEquals(vitalii.getEmail(), actualUser.getEmail());
+        Assertions.assertEquals(vitalii.getPassword(), actualUser.getPassword());
+    }
+
+    @Test
+    void login_OK() {
+        User expectedVitalii = new User();
+        String login = "vitalii@gmail.com";
+        String password = "1234";
+        Mockito.when(userService.findByEmail(expectedVitalii.getEmail()))
+                .thenReturn(Optional.of(expectedVitalii));
+        Mockito.when(passwordEncoder.matches(expectedVitalii.getPassword(), expectedVitalii.getPassword()))
+                .thenReturn(true);
+        try {
+            User actualVitalii = authenticationService.login(expectedVitalii.getEmail(),
+                    expectedVitalii.getPassword());
+            Assertions.assertNotNull(actualVitalii, "Actual user can not be null");
+            Assertions.assertEquals(expectedVitalii.getEmail(),actualVitalii.getEmail(), "Dude, e-mail is not correct!");
+            Assertions.assertEquals(expectedVitalii.getPassword(), actualVitalii.getPassword(),
+                    "Dude, you made mistake in password!");
+        } catch (AuthenticationException e) {
+            Assertions.assertEquals("Incorrect username or password!!!", e.getMessage());
+        }
+    }
+
+    @Test
+    void login_notExistentEmail_NotOk() {
+        User expectedVitalii = new User();
+        String login = "vitalii@gmail.com";
+        String password = "1234";
+        Mockito.when(userService.findByEmail(expectedVitalii.getEmail()))
+                .thenReturn(Optional.of(expectedVitalii));
+        Mockito.when(passwordEncoder.matches(expectedVitalii.getPassword(), expectedVitalii.getPassword()))
+                .thenReturn(true);
+        try {
+            User actualVitalii = authenticationService.login("vitali@gmail.com", "1234");
+            Assertions.assertNotNull(expectedVitalii.getEmail(), actualVitalii.getEmail());
+            Assertions.assertEquals(expectedVitalii.getPassword(), actualVitalii.getPassword());
+        } catch (AuthenticationException e) {
+            Assertions.assertEquals("Incorrect username or password!!!", e.getMessage());
+            return;
+        }
+        Assertions.fail("Expected to receive AuthenticationException");
```

Comment:
```suggestion
        Assertions.fail("Expected to receive AuthenticationException while login with not existent login");
```

It's better to make test fail messages more informative